src/Plutarch/Extra/StateThread.hs

```diff
@@ -59,7 +62,13 @@ uniqueStateTokenMint ::
   Term s (PValue keys amounts) ->
   Term s PBool
 uniqueStateTokenMint thisPolicy mint =
-  pvalueOf # mint # thisPolicy # pconstant "" #== pconstant 1
+  pisSingleton
```

Comment:
You can use `ptryFromSingleton` and check the token name - value pair so the the second check can be avoided.

Reply:
Good call, done.